server/datastore/mysql/query_results.go

```diff
@@ -0,0 +1,148 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/jmoiron/sqlx"
+)
+
+const (
+	// QueryResultRowLimit is the maximum number of rows that can be stored per query
+	QueryResultRowLimit = 1000
+)
+
+// SaveQueryResultRow saves a query result row to the datastore and returns number of rows inserted
+func (ds *Datastore) SaveQueryResultRows(ctx context.Context, rows []*fleet.ScheduledQueryResultRow) error {
```

Comment:
AFAICS this is only used in tests. Do we need it defined in the datastore interface? (It can be added as an utility method for tests).

Reply:
makes sense, i don't want to bloat the interface

Reply:
OK, I believe that's also the case for `QueryResultRows` and `DeleteQueryResultsForHost`.